### PR TITLE
WIP: centralized baseline experiment scaffolding

### PR DESCRIPTION
**Pull request by lguerdan**

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -135,23 +140,22 @@ def train_epoch(
             for p in model.parameters():
                 param_norm = p.grad.detach().data.norm(2)
                 total_norm += param_norm.item() ** 2
-            b += 1 if total_norm ** 0.5 < S else 0
+            b += 1 if total_norm ** 0.5 < S_e else 0
```
Comment (lguerdan) [inline train.py:143]: I think we should apply `S_e` (the value that was updated after the last batch), right? This also applies to 146. Another option is to use `S` from the previous epoch.

**Review by lguerdan — COMMENTED** [review]
When I ran the adaptive clipping, it looks like there are convergence issues where accuracy doesn't improve over training. This may be an issue in the code somewhere. A few possible causes could include: 

- `S` cutoff is monotonically increasing over epochs when we look at a plot, which seems fishy. It should also decrease, sometimes. I wonder how the distribution of `b` looks during training? 
- Maybe a lower `lr_c` parameter would stabilize training?

File: train.py [hunk 2]
```diff
@@ -135,23 +140,22 @@ def train_epoch(
...
             b += torch.randn(1) * sigma_b
```
Comment (jiachend) [inline train.py:144]: I think we can also move this line to line 155 and 158 so that the noise is only added for each batch

File: train.py (hunk 1 above)
Comment (jiachend) [reply train.py:143]: Right
